pkg/vsphere/vm/vm.go

```diff
@@ -69,12 +69,35 @@ func (vm *VirtualMachine) FolderName(ctx context.Context) (string, error) {
 }
 
 func (vm VirtualMachine) WaitForMAC(ctx context.Context) (map[string]string, error) {
+	devices, err := vm.Device(ctx)
+	if err != nil {
+		log.Errorf("Unable to get device listing for VM")
+		return nil, err
+	}
+
+	nics := devices.SelectByType(&types.VirtualEthernetCard{})
```

Comment:
add a comment about what is being mapped to what. I presume this is mapping an arbitrary label to "portgroupname" but only because of the variable names where this is used.